### PR TITLE
EVG-20708: add tests for setup.initial errors

### PR DESCRIPTION
**Pull request by Kimchelly**
EVG-20708

### Description
Add tests for `setup.initial` to ensure that it handles setup issues gracefully.

### Testing
Added unit tests.

### Documentation
N/A

### CODE REVIEW

File: agent/agent_test.go [hunk 1]
```diff
@@ -63,15 +66,20 @@ func TestAgentSuite(t *testing.T) {
 	suite.Run(t, new(AgentSuite))
 }
 
+// kim: TODO: fix merge conflict due to task working directory consolidation
```
Comment (julianedwards) [inline agent/agent_test.go:69]: Still have a TODO here.

File: agent/agent_test.go [hunk 2]
```diff
@@ -1234,25 +1239,152 @@ func (s *AgentSuite) TestFinishPrevTaskWithSameBuildButDifferentTaskGroup() {
 	s.Empty(taskDirectory)
 }
 
-func (s *AgentSuite) TestGeneralTaskSetupSucceeds() {
+// kim: TODO: fix gnarly merge conflict from task working directory PR.
```
Comment (julianedwards) [inline agent/agent_test.go:1242]: Another TODO.

Comment (hadjri) [inline agent/agent_test.go:1242]: Leftover debugging?

File: agent/agent_test.go [hunk 3]
```diff
@@ -63,15 +66,20 @@ func TestAgentSuite(t *testing.T) {
 	suite.Run(t, new(AgentSuite))
 }
 
+// kim: TODO: fix merge conflict due to task working directory consolidation
 func (s *AgentSuite) SetupTest() {
 	var err error
+	s.tmpDirName, err = os.MkdirTemp("", filepath.Base(s.T().Name()))
```
Comment (julianedwards) [inline agent/agent_test.go:72]: Nit: use `s.T().TempDir()`

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: agent/agent_test.go (hunk 1 above)
Comment (hadjri) [inline agent/agent_test.go:69]: Same question here

**Review by hadjri — COMMENTED** [review]
Tests LGTM

File: agent/agent_test.go (hunk 1 above)
Comment (Kimchelly) [reply agent/agent_test.go]: This was waiting on #7030 to be merged, so the merge conflict is gone now.

Comment (Kimchelly) [reply agent/agent_test.go]: This was waiting on #7030 to be merged, so the merge conflict is gone now.

File: agent/agent_test.go (hunk 2 above)
Comment (Kimchelly) [reply agent/agent_test.go]: Same as above.

Comment (Kimchelly) [reply agent/agent_test.go]: Same as above.

File: agent/agent_test.go (hunk 3 above)
Comment (Kimchelly) [reply agent/agent_test.go]: This isn't relevant anymore after merging #7030.

**Review by hadjri — APPROVED** [review]

**Review by julianedwards — APPROVED** [review]

**Pull request merged**